### PR TITLE
[Feature][Http] Http paramter support custom encryption

### PR DESCRIPTION
**Pull request by zhibinF**
### Purpose of this pull request

Added the ability to encrypt config files using http requests

### Does this PR introduce _any_ user-facing change?

Added the ability to encrypt config files using http requests

### How was this patch tested?

test it in RestAPiIT

### Check list

* [x] If necessary, please update the documentation to describe the new feature.

### CODE REVIEW

**Comment by zhibinF** [comment]
@liugddx @xiaofan2022 PTAL

**Comment by Hisoka-X** [comment]
cc @liugddx

**Comment by Hisoka-X** [comment]
Please resolve conflict. Thanks.

**Review by liugddx — COMMENTED** [review]
Is scalable encryption supported?

**Comment by zhibinF** [comment]
Let me check it.

**Comment by liugddx** [comment]
Please resolve the conflict.

**Comment by zhibinF** [comment]
> Please resolve the conflict.

done

**Comment by zhibinF** [comment]
> Is scalable encryption supported?

yes

File: docs/en/seatunnel-engine/rest-api.md [hunk 1]
```diff
@@ -262,3 +262,91 @@ network:
...
+<details>
+<summary><code>POST</code> <code><b>/hazelcast/rest/maps/encrypt-config</b></code> <code>(Returns the encrypted config if config is encrypted successfully.)</code></summary>
+
+#### Body
+
+```json
+{
+    "env": {
+        "execution.parallelism": 1,
+        "shade.identifier":"base64"
```
Comment (liugddx) [inline docs/en/seatunnel-engine/rest-api.md:279]: What other types of encryption do we support?  and base64 is not considered an encryption method and I can decrypt it easily.

Comment (zhibinF) [reply docs/en/seatunnel-engine/rest-api.md:279]: Base64 encryption and decryption are provided at now, and users can implement their own encryption and decryption.

Comment (liugddx) [reply docs/en/seatunnel-engine/rest-api.md:279]: > Base64 encryption and decryption are provided at now, and users can implement their own encryption and decryption.

Can we tell users in the documentation how to extend the encryption method?

Comment (zhibinF) [reply docs/en/seatunnel-engine/rest-api.md:279]: Yes, we have documentation. [Config File Encryption And Decryption](https://seatunnel.apache.org/docs/2.3.3/connector-v2/Config-Encryption-Decryption)

Comment (liugddx) [reply docs/en/seatunnel-engine/rest-api.md:279]: > Yes, we have documentation. [Config File Encryption And Decryption](https://seatunnel.apache.org/docs/2.3.3/connector-v2/Config-Encryption-Decryption)

Can you quote it in the document?  Users may not know how to customize encryption methods

Comment (zhibinF) [reply docs/en/seatunnel-engine/rest-api.md:279]: Yes, I will quote it tomorrow.

**Review by liugddx — DISMISSED** [review]

File: docs/en/seatunnel-engine/rest-api.md [hunk 2]
```diff
@@ -262,3 +262,91 @@ network:
...
 </details>
 
+------------------------------------------------------------------------------------------
+
+### Encrypt Config.
+
+<details>
+<summary><code>POST</code> <code><b>/hazelcast/rest/maps/encrypt-config</b></code> <code>(Returns the encrypted config if config is encrypted successfully.)</code></summary>
+For more information about customize encryption, please refer to the documentation [config-encryption-decryption](../connector-v2/Config-Encryption-Decryption.md).
+#### Body
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api.md:273]:
```suggestion

#### Body

```
image
The format have some problem.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 3]
```diff
@@ -253,6 +253,57 @@ public void testStartWithSavePointWithoutJobId() {
...
+                .post(
+                        HOST
+                                + hazelcastInstance
+                                        .getCluster()
+                                        .getLocalMember()
+                                        .getAddress()
+                                        .getPort()
+                                + RestConstant.ENCRYPT_CONFIG)
+                .then()
+                .statusCode(200);
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: Please assert response data.

**Review by Hisoka-X — COMMENTED** [review]
Thanks @zhibinF , just two small problem.

**Comment by Hisoka-X** [comment]
image
Please answer these question. Thanks.

File: docs/en/seatunnel-engine/rest-api.md (hunk 2 above)
Comment (zhibinF) [reply docs/en/seatunnel-engine/rest-api.md:273]: done.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java (hunk 3 above)
Comment (zhibinF) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: done.

**Comment by zhibinF** [comment]
> image Please answer these question. Thanks.

done.

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Please fix ci

**Review by liugddx — APPROVED** [review]

**Pull request merged**